rust/worker/src/execution/operators/filter.rs

```diff
@@ -231,6 +231,10 @@ impl<'me> MetadataProvider<'me> {
                     metadata_segment_reader.full_text_index_reader.as_ref(),
                     record_segment_reader,
                 ) {
+                    // The pattern can match empty string and thus match any document
+                    if let Some(0) = chroma_regex.properties().minimum_len() {
+                        return Ok(rec_reader.get_offset_stream(..).try_collect().await?);
```

Comment:
Can we TODO the perf optimization to avoid the scan here? We can just return a SignedRoaringBitmap